### ISSUE TITLE
Validate Workflow Examples with cwltool

### ISSUE THREAD

**Issue by tmooney** [genome/analysis-workflows#958]
I was working on another issue and discovered that the example input YAML I was using was outdated and the workflow I was running could be run with Cromwell but not `cwltool`.  Before putting out a new release, it'd be good to polish up the examples and make sure they can be run with `cwltool`.  (Our current automated tests are pinned to an older version; should we also test against the latest `cwltool` for the release?)

This is somewhat related to #645, but with a different objective.

**Comment by tmooney** [comment]
To clarify, this isn't only about `cwltool --validate`--there are some things that only come up at runtime.  (A bonus if we get some examples through with `cwltool` is that they could be useful for #116!)

**Comment by gschang** [comment]
I started working from the latest immuno workflow.

**Comment by gschang** [comment]
As new updates on testing with somatic_exome workflow examples, I finally got a succeeded build using the latest analysis-workflows Git repository (as of today, #1000 merged). My test job went smooth and I don't have any issue. All examples for somatic_exome workflow successfully work. Below are two points to discuss.


(1) Interpretation of results
I got only one somatic variant on TP53 as a result (ENSP00000269305.4:p.Arg175His missense mutation) from the current example sequencing data.

Hi, @chrisamiller. May I have your advice on this result interpretation? Do these results make sense to you? If you want to take a look at them, below is the results directory. Let me know if you cannot access that folder.

/gscuser/gchang/work2/14_BGA-WALKER/210203-cwltool2examples-958/1.analysis-workflows-test/example_data/test/results


(2) Next plan for aggregation

This test was done with my somatic_exome YAML update. But I didn't push it yet because I saw Chris's work on this in parallel--for example, VCFs for bqsr_known_sites and upgrade for per_target_intervals. I want to aggregate all these efforts to a final one.

Hi, @chrisamiller. Please let me know your thought and plan on somatic_exome test. I will find out what I can do more to finalize this somatic_exome workflow example. I would like to move on for other workflow examples--for example, rnaseq or bisulfite workflow (if you suggest). This is very exciting for me to get into and contribute to analysis workflows as Jason advised. I am happy to further discuss with you all.

**Comment by gschang** [comment]
Here is a progress update on Workflow Examples, especially somatic_exome CWL workflow. As updated above, I confirmed that the current examples work without any problem for the latest somatic_exome workflow (at analysis-workflows Git).

@johnegarza is now working on new examples for the somatic_exome workflow by cutting down to the BRCA genes, to replace the current examples. He is cleaning up huge example files (#645 Clean up example_data directory).

**Comment by gschang** [comment]
As the next workflow example, I am testing the latest rnaseq.cwl and rnaseq_star_fusion.cwl with the current workflow examples. I have updated the rnaseq YAML example (analysis-workflows/example_data/rnaseq/workflow.yaml) for test. Most reference inputs in the YAML files need updating with the latest ones we're using. My test job with rnaseq.cwl is running, in order to find out any broken steps if any, as I have seen them in somatic_exome.cwl.

**Comment by gschang** [comment]
I found that the current `definitions/pipelines/rnaseq.cwl` is broken after the **merge** step. Sridhar and I opened a new issue to fix this problem (#1006). In brief, we're adding a new step to sort a merged RNA-seq BAM between the `merge` and `index_bam` step.

**Comment by gschang** [comment]
I have submitted a new PR to fix the broken *rnaseq.cwl*. I merged PR #1007 to fix the problem in indexing a merged BAM. I have created the next PR #1013 to finalize this work.

I made the rnaseq.cwl workflow example up-to-date and fixed the gpbigwig_bamcoverage step, as I discussed with @sridhar0605 (#1007). In sum, I finally validated the current rnaseq.cwl, and it is working now with this updated workflow example.
